vscode-dotnet-runtime-library/src/Utils/CommandExecutor.ts

```diff
@@ -84,62 +107,277 @@ Please install the .NET SDK manually by following https://learn.microsoft.com/en
-        {
-            // The '.' character is not allowed for sudo-prompt so we use 'NET'
-            let sanitizedCallerName = this.context?.acquisitionContext?.requestingExtensionId?.replace(/[^0-9a-z]/gi, ''); // Remove non-alphanumerics per OS requirements
-            sanitizedCallerName = sanitizedCallerName?.substring(0, 69); // 70 Characters is the maximum limit we can use for the prompt.
-            const options = { name: `${sanitizedCallerName ?? '.NET Install Tool'}` };
+        const oldReturnStatusSetting = this.returnStatus;
+        this.returnStatus = true;
+        const masterSudoProcessSpawnResult = this.startupSudoProc(fullCommandString, shellScript, terminalFailure);
+        this.returnStatus = oldReturnStatusSetting;
+
+        await this.sudoProcIsLive(terminalFailure);
+        return this.executeSudoViaProcessCommunication(fullCommandString, terminalFailure);
+    }
 
-            this.context?.eventStream.post(new CommandExecutionUserAskDialogueEvent(`Prompting user for command ${fullCommandString} under sudo.`));
-            exec((fullCommandString), options, (error?: any, stdout?: any, stderr?: any) =>
+    /**
+     *
+     * @param fullCommandString the command that will be run by the master process once it is spawned, not super relevant here, used for logging.
+     * @param shellScriptPath the path of the shell script file for the process to run that should loop and follow the protocol procedure
+     * @param terminalFailure whether if we cannot start the sudo process, should we fail the entire program.
+     * @returns The string result of either trying to spawn the sudo master process, or the status code of that attempt depending on the return mode.
+     */
+    private async startupSudoProc(fullCommandString : string, shellScriptPath : string, terminalFailure : boolean) : Promise<string>
+    {
+        if(this.hasEverLaunchedSudoFork)
+        {
+            if(await this.sudoProcIsLive(false))
             {
-                let commandResultString = '';
+                return Promise.resolve('0');
+            }
+        }
+        this.hasEverLaunchedSudoFork = true;
 
-                if (stdout)
-                {
-                    this.context?.eventStream.post(new CommandExecutionStdOut(`The command ${fullCommandString} encountered stdout, continuing
+        // Launch the process under sudo
+        this.context?.eventStream.post(new CommandExecutionUserAskDialogueEvent(`Prompting user for command ${fullCommandString} under sudo.`));
+
+        // The '.' character is not allowed for sudo-prompt so we use 'NET'
+        let sanitizedCallerName = this.context?.acquisitionContext?.requestingExtensionId?.replace(/[^0-9a-z]/gi, ''); // Remove non-alphanumerics per OS requirements
+        sanitizedCallerName = sanitizedCallerName?.substring(0, 69); // 70 Characters is the maximum limit we can use for the prompt.
+        const options = { name: `${sanitizedCallerName ?? '.NET Install Tool'}` };
```

Comment:
Isn't the `.` in `.NET Install Tool` invalid per the comment above?

Reply:
Very good call, thank you.